internal/cluster/transition_test.go

```diff
@@ -330,6 +330,8 @@ type validationCheckResult struct {
 	messagePattern string
 }
 
+type candidateChecker func()
```

Comment:
No need to define global if it's used only inside one function, you can probably define it inside.